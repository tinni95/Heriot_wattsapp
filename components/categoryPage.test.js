import CategoryPage from "./categoryPage";
import renderer from "react-test-renderer";
import React from "react";

const propsOne = {
    search: "",
    menuItems: {
        edges: [
            {
                node: {
                    name: "mangia",
                    price: 20,
                },
            },
            {
                node: {
                    name: "mangiaa",
                    price: 10,
                },
            },
        ],
    },
};

const propsTwo = {
    ...propsOne,
    subCategories: {
        edges: [
            {
                node: {
                    name: "Panini",
                    menuItems: {
                        edges: [
                            {
                                node: {
                                    name: "cacca",
                                    price: 20,
                                },
                            },
                        ],
                    },
                },
            },
        ],
    },
};

const propsThree = {
    ...propsOne,
    search: "cascaca",
};

describe("categoryPage", () => {
    describe("without subCategories", () => {
        let rendered;
        beforeEach(() => {
            rendered = renderer.create(<CategoryPage {...propsOne} />);
        });
        it("renders without subcategories", () => {
            expect(rendered).toBeTruthy();
        });
    });

    describe("with subCategories", () => {
        let rendered, itemCards;
        beforeEach(() => {
            rendered = renderer.create(<CategoryPage {...propsTwo} />);
        });
        it("renders without subcategories", () => {
            expect(rendered).toBeTruthy();
        });
    });
});
