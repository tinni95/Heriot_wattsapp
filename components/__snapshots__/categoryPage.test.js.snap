// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`categoryPage with subCategories matches snapshot 1`] = `
<View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "marginTop": 20,
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#F4F4F4",
          "borderRadius": 10,
          "flex": 1,
          "flexDirection": "row",
          "height": 100,
          "margin": 8,
          "width": 320,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 6,
            "flexDirection": "column",
            "margin": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 24,
              "fontWeight": "bold",
            }
          }
        >
          mangia
        </Text>
        <Text
          style={
            Object {
              "fontSize": 16,
              "marginTop": 15,
            }
          }
        >
          £
          20
        </Text>
      </View>
      <View
        style={
          Object {
            "alignContent": "center",
            "backgroundColor": "#348653",
            "borderBottomRightRadius": 10,
            "borderTopRightRadius": 10,
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "fontSize": 20,
              "textAlign": "center",
            }
          }
        >
          +
        </Text>
      </View>
    </View>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "marginTop": 20,
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#F4F4F4",
          "borderRadius": 10,
          "flex": 1,
          "flexDirection": "row",
          "height": 100,
          "margin": 8,
          "width": 320,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 6,
            "flexDirection": "column",
            "margin": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 24,
              "fontWeight": "bold",
            }
          }
        >
          mangiaa
        </Text>
        <Text
          style={
            Object {
              "fontSize": 16,
              "marginTop": 15,
            }
          }
        >
          £
          10
        </Text>
      </View>
      <View
        style={
          Object {
            "alignContent": "center",
            "backgroundColor": "#348653",
            "borderBottomRightRadius": 10,
            "borderTopRightRadius": 10,
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "fontSize": 20,
              "textAlign": "center",
            }
          }
        >
          +
        </Text>
      </View>
    </View>
  </View>
  <View>
    <Text
      style={
        Object {
          "fontSize": 25,
          "fontWeight": "bold",
          "margin": 25,
        }
      }
    >
      Panini
    </Text>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "marginTop": 20,
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#F4F4F4",
            "borderRadius": 10,
            "flex": 1,
            "flexDirection": "row",
            "height": 100,
            "margin": 8,
            "width": 320,
          }
        }
      >
        <View
          style={
            Object {
              "flex": 6,
              "flexDirection": "column",
              "margin": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 24,
                "fontWeight": "bold",
              }
            }
          >
            cacca
          </Text>
          <Text
            style={
              Object {
                "fontSize": 16,
                "marginTop": 15,
              }
            }
          >
            £
            20
          </Text>
        </View>
        <View
          style={
            Object {
              "alignContent": "center",
              "backgroundColor": "#348653",
              "borderBottomRightRadius": 10,
              "borderTopRightRadius": 10,
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#FFFFFF",
                "fontSize": 20,
                "textAlign": "center",
              }
            }
          >
            +
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`categoryPage without subCategories matches snapshot 1`] = `
<View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "marginTop": 20,
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#F4F4F4",
          "borderRadius": 10,
          "flex": 1,
          "flexDirection": "row",
          "height": 100,
          "margin": 8,
          "width": 320,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 6,
            "flexDirection": "column",
            "margin": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 24,
              "fontWeight": "bold",
            }
          }
        >
          mangia
        </Text>
        <Text
          style={
            Object {
              "fontSize": 16,
              "marginTop": 15,
            }
          }
        >
          £
          20
        </Text>
      </View>
      <View
        style={
          Object {
            "alignContent": "center",
            "backgroundColor": "#348653",
            "borderBottomRightRadius": 10,
            "borderTopRightRadius": 10,
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "fontSize": 20,
              "textAlign": "center",
            }
          }
        >
          +
        </Text>
      </View>
    </View>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "marginTop": 20,
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#F4F4F4",
          "borderRadius": 10,
          "flex": 1,
          "flexDirection": "row",
          "height": 100,
          "margin": 8,
          "width": 320,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 6,
            "flexDirection": "column",
            "margin": 10,
          }
        }
      >
        <Text
          style={
            Object {
              "fontSize": 24,
              "fontWeight": "bold",
            }
          }
        >
          mangiaa
        </Text>
        <Text
          style={
            Object {
              "fontSize": 16,
              "marginTop": 15,
            }
          }
        >
          £
          10
        </Text>
      </View>
      <View
        style={
          Object {
            "alignContent": "center",
            "backgroundColor": "#348653",
            "borderBottomRightRadius": 10,
            "borderTopRightRadius": 10,
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "fontSize": 20,
              "textAlign": "center",
            }
          }
        >
          +
        </Text>
      </View>
    </View>
  </View>
</View>
`;
